src/modules/currency/formatAmountByParts.ts

```diff
@@ -0,0 +1,54 @@
+import { CURRENCIES } from './data/currencies';
+import { ByParts } from './types';
+import { withErrorBoundary } from '../../common/errorBoundary';
+import { getLocale } from '../shared/utils';
+
+const formatAmountByParts = (
+  currencyCode: keyof typeof CURRENCIES,
+  amount: string | number,
+  locale?: string,
+): ByParts => {
+  if (!Number(amount) && Number(amount) !== 0)
+    throw new Error('Parameter `amount` is not a number!');
+
+  // If a specific locale is provided, use it; otherwise, use the browser's locale
+  if (!locale) {
```

Comment:
have a shared utility between formatNumber and formatNumberByParts that re-uses the logic before passing it to Intl. 
This logic will contain computing locale, deciding currency inputs for Intl, etc.